### ISSUE TITLE
Creating a new static webapp fails

### ISSUE THREAD

**Issue by JacobStruiksma** [Azure/azure-powershell#16594]
### Description

When creating a new static webapp in a existsing resource group fails

### Issue script & Debug output

```PowerShell
PS> $DebugPreference='Continue'
PS> New-AzStaticWebApp -Name 'powershellstatiswebtest' -ResourceGroupName 'ResourceGroupName' -Location 'West Europe' -SkuName 'Standard'

DEBUG: Registering Az shared AssemblyLoadContext for path: 'C:\Users\xxx\Documents\PowerShell\Modules\Az.Accounts\2.7.0\StartupScripts\..\AzSharedAlcAssemblies'.
DEBUG: AssemblyLoadContext registered.
DEBUG: 16:27:54 - GetAzureRMContextCommand end processing.
DEBUG: [CmdletBeginProcessing]: Starting command
DEBUG: CmdletBeginProcessing:    
DEBUG: CmdletProcessRecordStart: 
DEBUG: [CmdletProcessRecordAsyncStart]: Created new QosEvent for command 'New-AzStaticWebApp_CreateExpanded'
DEBUG: CmdletProcessRecordAsyncStart: 
DEBUG: CmdletGetPipeline: 
DEBUG: CmdletBeforeAPICall: 
DEBUG: URLCreated: /subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/staticSites/powershellstatiswebtest?api-version=2020-12-01    
DEBUG: RequestCreated: /subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/staticSites/powershellstatiswebtest?api-version=2020-12-01
DEBUG: HeaderParametersAdded: 
DEBUG: BodyContentSet: 
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{ResourceGroupName}/providers/Microsoft.Web/staticSites/powershellstatiswebtest?api-version=2020-12-01

Headers:
x-ms-unique-id                : 1,2,3
x-ms-client-request-id        : f0616853-cc27-4b80-885f-a232f5388fbc
CommandName                   : New-AzStaticWebApp
FullCommandName               : New-AzStaticWebApp_CreateExpanded
ParameterSetName              : __AllParameterSets
User-Agent                    : AzurePowershell/v0.0.0,Az.Websites/0.0.0

Body:
{
  "location": "West Europe",
  "sku": {
    "name": "Standard"
  }
}


DEBUG: BeforeCall:
DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Cache-Control                 : no-cache
Pragma                        : no-cache
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Server                        : Microsoft-IIS/10.0
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-request-id               : 6804933a-87c6-4fe0-ad65-febbfe7edca4
x-ms-correlation-request-id   : 6804933a-87c6-4fe0-ad65-febbfe7edca4
x-ms-routing-request-id       : WESTEUROPE:20211207T152817Z:6804933a-87c6-4fe0-ad65-febbfe7edca4
X-Content-Type-Options        : nosniff
Date                          : Tue, 07 Dec 2021 15:28:16 GMT

Body:
{
  "Code": "BadRequest",
  "Message": "Request Envelope is invalid.  Please ensure the payload is not empty",
  "Target": null,
  "Details": [
    {
      "Message": "Request Envelope is invalid.  Please ensure the payload is not empty"
    },
    {
      "Code": "BadRequest"
    },
    {
      "ErrorEntity": {
        "ExtendedCode": "51021",
        "MessageTemplate": "{0} is invalid.  {1}",
        "Parameters": [
          "Request Envelope",
          "Please ensure the payload is not empty"
        ],
        "Code": "BadRequest",
        "Message": "Request Envelope is invalid.  Please ensure the payload is not empty"
      }
    }
  ],
  "Innererror": null
}


DEBUG: ResponseCreated:
DEBUG: BeforeResponseDispatch:
New-AzStaticWebApp_CreateExpanded: C:\Users\xxx\Documents\PowerShell\Modules\Az.Websites\2.9.0\Websites.Autorest\exports\ProxyCmdletDefinitions.ps1:5975:9
Line |
5975 |          $steppablePipeline.Begin($PSCmdlet)
     |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The server responded with a Request Error, Status: BadRequest

DEBUG: [Finally]: Getting exception 'Microsoft.Azure.Commands.Common.Exceptions.AzPSCloudException: InternalException' from response
DEBUG: Finally: 
DEBUG: CmdletAfterAPICall:
DEBUG: [CmdletProcessRecordAsyncEnd]: Finish HTTP process
DEBUG: CmdletProcessRecordAsyncEnd:
DEBUG: CmdletProcessRecordEnd:
DEBUG: AzureQoSEvent: Module: Az.Websites:0.0.0; CommandName: New-AzStaticWebApp_CreateExpanded; PSVersion: ; IsSuccess: False; Duration: 00:00:00; Exception: InternalException;
DEBUG: Finish sending metric.
DEBUG: CmdletEndProcessing:
```

### Environment data

```PowerShell
PS > $PSVersionTable 

Name                           Value
----                           -----
PSVersion                      7.2.0
PSEdition                      Core
GitCommitId                    7.2.0
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Module versions

```PowerShell
PS > Get-Module Az*


ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Script     2.7.0                 Az.Accounts                         {Add-AzEnvironment, Clear-AzContext, Clear-AzDefault, Connect-AzAccount…}
Script     2.9.0                 Az.Websites                         {Add-AzWebAppAccessRestrictionRule, Add-AzWebAppTrafficRouting, Edit-AzWebAppBackupConfiguration, Enter-AzWebAppContainerPSSession…}
```

### Error output

```PowerShell
PS > Resolve-AzError

DEBUG: Sought all Az modules and got latest version 7.0.0
DEBUG: 16:32:44 - ResolveError begin processing with ParameterSet 'AnyErrorParameterSet'.
DEBUG: 16:32:44 - using account id '{UPN}'...
WARNING: Upcoming breaking changes in the cmdlet 'Resolve-AzError' :
The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use `Resolve-AzError` instead.
Note : Go to https://aka.ms/azps-changewarnings for steps to suppress this breaking change warning, and other information on breaking changes in Azure PowerShell.

   HistoryId: 1

Message        : The server responded with a Request Error, Status: BadRequest
StackTrace     : 
Exception      : Microsoft.Azure.PowerShell.Cmdlets.Websites.Runtime.RestException`1[Microsoft.Azure.PowerShell.Cmdlets.Websites.Models.Api20201201.IDefaultErrorResponse]
InvocationInfo : {New-AzStaticWebApp_CreateExpanded}
Line           :         $steppablePipeline.Begin($PSCmdlet)

Position       : At C:\Users\jstruiksma\Documents\PowerShell\Modules\Az.Websites\2.9.0\Websites.Autorest\exports\ProxyCmdletDefinitions.ps1:5975 char:9
                 +         $steppablePipeline.Begin($PSCmdlet)
                 +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
HistoryId      : 1


The Azure PowerShell team is listening, please let us know how we are doing: https://aka.ms/azpssurvey?Q_CHL=ERROR.

DEBUG: AzureQoSEvent: Module: Az.Accounts:2.7.0; CommandName: Resolve-AzError; PSVersion: 7.2.0; IsSuccess: True; Duration: 00:00:00.2034879
DEBUG: Finish sending metric.
DEBUG: 16:32:45 - ResolveError end processing.
```

**Comment by JacobStruiksma** [comment]
When adding the repositoryurl with the value $null the static webapp could succesfully be created
```
New-AzStaticWebApp -Name 'powershellstatiswebtest' -ResourceGroupName 'ResourceGroupName' -Location 'West Europe' -SkuName 'Standard' -RepositoryUrl $null
```

**Comment by LucasYao93** [comment]
Hi @JacobStruiksma,
[Here](https://docs.microsoft.com/en-us/powershell/module/az.websites/new-azstaticwebapp?view=azps-7.0.0#example-1--create-a-new-static-site-in-an-existing-resource-group--or-updates-an-existing-static-site) is eample of the New-AzStaticWebApp. The example includes how to deploy static webapp with git repo. Please try again. 
Please ping me if you have any questions.

Thanks,
Lucas

**Comment by LucasYao93** [comment]
Hi @dingmeng-xue 

Regarding the generated cmdlet [New-AzStaticWebApp (Az.Websites)](https://docs.microsoft.com/en-us/powershell/module/az.websites/new-azstaticwebapp?view=azps-7.0.0), some parameters are required. Because the parameters are defined as optional in swagger. Therefore, it becomes optional after generated.

Do we need to keep same with swaggers or modify them?

**Comment by JacobStruiksma** [comment]
So the real problem is that the properties field should be in the body but does not have to contain any values:
current body that does not work:
```
Body:
{
  "location": "West Europe",
  "sku": {
    "name": "Standard"
  }
}
```
Body that does work:

```
Body:
{
  "location": "West Europe",
  "sku": {
    "name": "Standard"
  },
  "properties": { }
}
```

**Comment by LucasYao93** [comment]
@JacobStruiksma,
Thanks for your information.
Body that does work:
```json
{
  "location": "West Europe",
  "properties": {
    "repositoryUrl": ""
  },
  "sku": {
    "name": "Standard"
  }
}
```
The `location`, `sku`, `repositoryUrl` are required.
Required `repositoryUrl` property is missing in the swagger. Therefore the `repositoryUrl` is optional in the `New-AzStaticWebApp` cmdlet. We have assigned [issue](https://github.com/Azure/azure-rest-api-specs/issues/17073) to the swagger team for fix it.

**Comment by LucasYao93** [comment]
> @JacobStruiksma, Thanks for your information. Body that does work:
> 
> ```json
> {
>   "location": "West Europe",
>   "properties": {
>     "repositoryUrl": ""
>   },
>   "sku": {
>     "name": "Standard"
>   }
> }
> ```
> 
> The `location`, `sku`, `repositoryUrl` are required. Required `repositoryUrl` property is missing in the swagger. Therefore the `repositoryUrl` is optional in the `New-AzStaticWebApp` cmdlet. We have assigned [issue](https://github.com/Azure/azure-rest-api-specs/issues/17073) to the swagger team for fix it.

Sorry, My mistake, The body must containes properties and the properties can be empty.

**Comment by LucasYao93** [comment]
Hi @JacobStruiksma 
We need to discuss how to fix this bug.

There is a workaround. Assgin empty string to `RepositoryUrl` when use  `New-AzStaticWebApp` to creating static web app.
After can use `New-AzStaticWebApp` to update `RepositoryUrl` if `RepositoryUrl` need to modify.

**Comment by Kotasudhakarreddy** [comment]
It requires a swagger change to fix this issue as few of the mandatory parameters were not made required in the swagger definition. Until we use the cmdlet refers to the updated swagger definition, please refer [here](https://github.com/Azure/azure-powershell/issues/16594#issuecomment-994241368) to unblock.

**Comment by Kotasudhakarreddy** [comment]
@panchagnula can you please include any one from API Swagger team.

**Comment by panchagnula** [comment]
@Kotasudhakarreddy I don't own SWA PS commands - so please don't tag on these. These commands are auto generated by PS team - let them attend to these. I don't have any contact on SWA team to help. @dingmeng-xue please take a look & contact the right team as needed. My team doesn't own SWA PS support.

**Comment by dingmeng-xue** [comment]
Hi Sisira, I know. Bot sends message just when I changed the label. I marked with "service" and "service attention" label means it requires the fix on service side or api spec.

**Comment by Kotasudhakarreddy** [comment]
We are closing this as the issue could not be reproduced with the most recent `Az.Websites` module version `2.12.1`. Please create a new issue if you are still facing it with latest version.

cc- @NitashaV
